js/modules/k6/http/http.go

```diff
@@ -21,101 +21,153 @@
-	OCSP_STATUS_UNKNOWN                string `js:"OCSP_STATUS_UNKNOWN"`
-	OCSP_REASON_UNSPECIFIED            string `js:"OCSP_REASON_UNSPECIFIED"`
-	OCSP_REASON_KEY_COMPROMISE         string `js:"OCSP_REASON_KEY_COMPROMISE"`
-	OCSP_REASON_CA_COMPROMISE          string `js:"OCSP_REASON_CA_COMPROMISE"`
-	OCSP_REASON_AFFILIATION_CHANGED    string `js:"OCSP_REASON_AFFILIATION_CHANGED"`
-	OCSP_REASON_SUPERSEDED             string `js:"OCSP_REASON_SUPERSEDED"`
-	OCSP_REASON_CESSATION_OF_OPERATION string `js:"OCSP_REASON_CESSATION_OF_OPERATION"`
-	OCSP_REASON_CERTIFICATE_HOLD       string `js:"OCSP_REASON_CERTIFICATE_HOLD"`
-	OCSP_REASON_REMOVE_FROM_CRL        string `js:"OCSP_REASON_REMOVE_FROM_CRL"`
-	OCSP_REASON_PRIVILEGE_WITHDRAWN    string `js:"OCSP_REASON_PRIVILEGE_WITHDRAWN"`
-	OCSP_REASON_AA_COMPROMISE          string `js:"OCSP_REASON_AA_COMPROMISE"`
+// Exports returns the JS values this module exports.
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Default: mi.exports,
+		// TODO: add new HTTP APIs like Client, Request, etc. as named exports?
+	}
+}
 
-	responseCallback func(int) bool
+// TODO: deprecate these?
+func (mi *ModuleInstance) defineConstants() {
+	rt := mi.vu.Runtime()
+	mustAddProp := func(name, val string) {
+		err := mi.exports.DefineDataProperty(
+			name, rt.ToValue(val), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE,
+		)
+		if err != nil {
+			common.Throw(rt, err)
+		}
+	}
+	mustAddProp("TLS_1_0", netext.TLS_1_0)
+	mustAddProp("TLS_1_1", netext.TLS_1_1)
+	mustAddProp("TLS_1_2", netext.TLS_1_2)
+	mustAddProp("TLS_1_3", netext.TLS_1_3)
+	mustAddProp("OCSP_STATUS_GOOD", netext.OCSP_STATUS_GOOD)
+	mustAddProp("OCSP_STATUS_REVOKED", netext.OCSP_STATUS_REVOKED)
+	mustAddProp("OCSP_STATUS_SERVER_FAILED", netext.OCSP_STATUS_SERVER_FAILED)
+	mustAddProp("OCSP_STATUS_UNKNOWN", netext.OCSP_STATUS_UNKNOWN)
+	mustAddProp("OCSP_REASON_UNSPECIFIED", netext.OCSP_REASON_UNSPECIFIED)
+	mustAddProp("OCSP_REASON_KEY_COMPROMISE", netext.OCSP_REASON_KEY_COMPROMISE)
+	mustAddProp("OCSP_REASON_CA_COMPROMISE", netext.OCSP_REASON_CA_COMPROMISE)
+	mustAddProp("OCSP_REASON_AFFILIATION_CHANGED", netext.OCSP_REASON_AFFILIATION_CHANGED)
+	mustAddProp("OCSP_REASON_SUPERSEDED", netext.OCSP_REASON_SUPERSEDED)
+	mustAddProp("OCSP_REASON_CESSATION_OF_OPERATION", netext.OCSP_REASON_CESSATION_OF_OPERATION)
+	mustAddProp("OCSP_REASON_CERTIFICATE_HOLD", netext.OCSP_REASON_CERTIFICATE_HOLD)
+	mustAddProp("OCSP_REASON_REMOVE_FROM_CRL", netext.OCSP_REASON_REMOVE_FROM_CRL)
+	mustAddProp("OCSP_REASON_PRIVILEGE_WITHDRAWN", netext.OCSP_REASON_PRIVILEGE_WITHDRAWN)
+	mustAddProp("OCSP_REASON_AA_COMPROMISE", netext.OCSP_REASON_AA_COMPROMISE)
 }
 
-// XCookieJar creates a new cookie jar object.
-func (*HTTP) XCookieJar(ctx *context.Context) *HTTPCookieJar {
-	return newCookieJar(ctx)
+func (mi *ModuleInstance) newCookieJar(call goja.ConstructorCall, rt *goja.Runtime) *goja.Object {
```

Comment:
```suggestion
func (mi *ModuleInstance) newCookieJar(call goja.ConstructorCall) *goja.Object {
```
:thinking: Is this expected to be a constructor? Is this form accepted, I'm not seeing it in the doc.

Reply:
maybe [this](https://pkg.go.dev/github.com/dop251/goja#readme-native-constructors) will help or do you mean in the `k6/http` [doc](https://k6.io/docs/javascript-api/k6-http/cookiejar) ?

Reply:
ok, so `goja` has two ways for defining a constructor:

```
func(ConstructorCall, *Runtime) *Object
```
or
```
func(ConstructorCall) *Object
```
@MStoykov Which one should we prefer in this case where both could work? Is the first one better in terms of performance? I need to check it in the meanwhile but maybe you already know the response.

Reply:
I don't think there is any performance difference, the first one just gives you access to the `Runtime` which we already have though, so no need to use that one :)

Reply:
In the second version we should use the `mi.vu.Runtime()` (to be get from the context) call for resolving the runtime so it means one more call, for that reason I was guessing that the first (already used) version would be faster.

Reply:
in that sense - yes @codebien probably using the first one is a bit faster, but IMO as we use the `mi.vu` to get everything else and it already has the runtime it makes more sense to just use that. 

>(to be get from the context) 

this is currently an implementation detail that will change, just the theoretical slowness of that probably doesn't matter for most cases while the fix for that will require quite a bit of reworking of `js` module ... last I looked at it.

So in retrospect you are correct, but IMO it's better to just get the runtime from `mi.vu` everywhere including in constructors where it's possible for goja to give it to us.

Reply:
Hmm do you also want me to change this? https://github.com/grafana/k6/blob/8a475aba3b2f096509f7d0f41752f0a02738f7ef/js/modules/k6/http/http.go#L165

Reply:
Assumed yes, so fixed both in https://github.com/grafana/k6/compare/8a475aba3b2f096509f7d0f41752f0a02738f7ef..2d45769d19a3fa08c9b1aa0a0340c8a4f63f81a0 :sweat_smile: